managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -111,7 +111,8 @@ public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
                 callback.readEntriesComplete(entries, ctx);
                 recycle();
             });
-        } else if (cursor.config.isAutoSkipNonRecoverableData() && exception instanceof NonRecoverableLedgerException) {
+        } else if (cursor.getManagedLedger().getConfig().isAutoSkipNonRecoverableData()
```

Comment:
`cursor.config` equals `getManagedLedger().getConfig()`.

When the topic is loaded, `org.apache.pulsar.broker.service.persistent.PersistentTopic#checkPersistencePolicies` will be called, which changes the ledger config, but the `cursor.config` will not be changed, so we need to fix that to get the latest config. 

BTW, the `cursor.conf` should be deleted, and always call the `getManagedLedger().getConfig()`.